[All] Documentation & Standards (#364)
manala/manala-recipes: lazy.ansible/README.md

```shell
make sh PORT=1234:4321
```

### Makefile

One of the first directive of your project's Makefile should be to include manala recipe `Makefile`
```makefile
include .manala/Makefile
```

> Or, if you're not sure manala recipe directory will be present, make it optional. This has the consequences of ensuring yourself that every parts of you project's Makefile using manala recipe tools could fallback to another solutions.

Reviewer: ```suggestion
Or, if you're not sure manala recipe directory will be present, make it optional. If you do make it optionnal, you'll have to ensure that every parts of you project's Makefile using manala recipe tools could fallback to another solutions.
```

```makefile
-include .manala/Makefile
```

Note: it's a good practise to have a `.SILENT:` as the first line of your project's Makefile. Not only does it globally silence make targets echoing, but it also offers a lowcost debugging system by just commenting it on demand.
```makefile
.SILENT:
```
```makefile
#.SILENT:
```

> A `MANALA_DOCKER` variable is available to check wether you're inside container or not

Reviewer: ```suggestion
A `MANALA_DOCKER` variable is available to check whether you're inside the container or not
```

```makefile
foo:
ifdef MANALA_DOCKER
	echo Inside container
else
	echo Outside container
endif
```

A `MANALA_DIR` variable is available to get your project's directory *inside* container
```makefile
foo:
	echo Project directory is $(MANALA_DIR)
```

There are three ways to force your project's Makefile target execution inside container:

1. Function `manala_docker_shell`

```makefile
foo:
	echo Can be run *inside* or *outside* container
	$(call manala_docker_shell, echo Always run *inside* container)
```

2. Shell substitution with `MANALA_DOCKER_SHELL`

```makefile
foo: SHELL := $(MANALA_DOCKER_SHELL)
foo:
	echo Always run *inside* container

# Or, if you're not sure manala recipe "Makefile" will be included
foo: SHELL := $(or $(MANALA_DOCKER_SHELL),$(SHELL))
foo:
	echo Can be run *inside* or *outside* container, \
		but always *inside* if manala recipe "Makefile" is included
```

3. Make substitution with `MANALA_DOCKER_MAKE`

```makefile
foo:
	$(MANALA_DOCKER_MAKE) bar

bar:
	echo Run *outside* container if called directly \
		or *inside* container if called from "foo"
```

> An automagic command help system is included out-of-the-box. You can list all available documented commands with `make help` (or just `make`,  as the help command is the default one)

Reviewer: ```suggestion
An automagic target help system is included out-of-the-box. You can list all available documented commands with `make help` (or just `make`,  as the help command is the default one)
```
